sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -49,8 +49,19 @@ case object AllTuples extends Distribution
  * [[Expression Expressions]] will be co-located. Based on the context, this
  * can mean such tuples are either co-located in the same partition or they will be contiguous
  * within a single partition.
+ * There is also another constraint, the `clustering` value contains null will be considered
+ * as a valid value if `nullKeysSensitive` == true.
+ *
+ * For examples:
+ * JOIN KEYS: values contains null will be considered as invalid values, which means
```

Comment:
Should be the input value. (Either the original data from table or the intermediate result(e.g. join outputs)).

Validity of the null in the original table, depends on the semantics, in `Join`, it's should also be invalid, but it's valid for `Group BY`.

It would an other optimization for repartition, contains `null` in the join keys.